cmd/nodecmd/create_devnet.go

```diff
@@ -188,16 +188,41 @@ func setupDevnet(clusterName string) error {
 			return err
 		}
 	}
-
 	// update node/s genesis + conf and start
-	if err := ansible.RunAnsiblePlaybookSetupDevnet(
-		app.GetAnsibleDir(),
-		strings.Join(ansibleHostIDs, ","),
-		app.GetNodesDir(),
-		app.GetAnsibleInventoryDirPath(clusterName),
-	); err != nil {
+	hosts, err := ansible.GetInventoryFromAnsibleInventoryFile(inventoryPath)
+	if err != nil {
 		return err
 	}
+	wg := sync.WaitGroup{}
+	wgResults := models.NodeResults{}
+	for _, host := range hosts {
+		wg.Add(1)
+		go func(nodeResults *models.NodeResults, host models.Host) {
+			defer wg.Done()
+			if err := host.Connect(constants.SSHScriptTimeout); err != nil {
+				nodeResults.AddResult(host.NodeID, nil, err)
+				return
+			}
+			defer func() {
+				if err := host.Disconnect(); err != nil {
+					nodeResults.AddResult(host.NodeID, nil, err)
+				}
+			}()
+			keyPath := filepath.Join(app.GetNodesDir(), host.GetCloudID())
+			if err := ssh.RunSSHSetupDevNet(host, keyPath); err != nil {
+				nodeResults.AddResult(host.NodeID, nil, err)
+				return
+			}
+		}(&wgResults, host)
+	}
+	wg.Wait()
+	for _, node := range hosts {
+		if wgResults.HasNodeIDWithError(node.NodeID) {
+			ux.Logger.PrintToUser("Node %s is ERROR with error: %s", node.NodeID, wgResults.GetErroHostMap()[node.NodeID])
+		} else {
+			ux.Logger.PrintToUser("Node %s is SETUP as devnet", node.NodeID)
+		}
+	}
```

Comment:
can we stop configuration and return error is some node was not setup?

Reply:
yes